### DISCUSSION TITLE
Rust-analyzer does not apply on src/bin and tests folder

### DISCUSSION THREAD

**Discussion by perfogic** [rust-lang/rust-analyzer#16481] [Q&A]
Hello, i face the above error. I can see the description of each function and do navigation to implementation of function on import module. But it does not work on src/bin and tests folder. How to fix this ?
[image that rust-analyzer works](https://i.stack.imgur.com/vxnCu.png)
[image that rust-analyzer does not work](https://i.stack.imgur.com/MnC0v.png)

Tree structure:
.
├── bin
├── build.rs
├── Cargo.lock
├── Cargo.toml
├── src
├── tests
└── wasm

Tree structure in src:
├── src
│   ├── app
│   ├── app.rs
│   ├── bin
│   ├── bitcoin
│   ├── constants.rs
│   ├── cosmos.rs
│   ├── error.rs
│   ├── lib.rs
│   ├── network.rs
│   └── utils.rs
Thanks for your time!

**Comment by alibektas** [comment]
Based on the folder names it is likely that you have your project structure wrong but if you really want to use these folder I guess you need to `mod bin` it in your main.rs or lib.rs.

**Reply by perfogic** [reply]
Thank you man, and how to add tests folder ? Thank you. I have updated the tree structure of my project

**Reply by alibektas** [reply]
Based on your recent addings everything seems fine to me. I suggest you read more about  chapter about testing in the rust book, and cargo keywords such as `include`.

**Reply by alibektas** [reply]
And I take back what I said before about using `pub mod` since you moved these folder outside of src folder now

**Reply by perfogic** [reply]
Can you give me the link ? Thanks for your time!

**Reply by perfogic** [reply]
I have read https://doc.rust-lang.org/book/ch11-03-test-organization.html. It show me about tests folder but don't see any suggestions about this

**Reply by alibektas** [reply]
It's odd that you couldn't get rust analyzer to work with your current project structure. I would like to help but I need to see more, maybe you can share your repo if it's publicly available.

**Reply by perfogic** [reply]
sure man, thank you a lot. Here is the repo: https://github.com/nomic-io/nomic

**Reply by alibektas** [reply]
no worries. I've just checked if it works on the said repo and I haven't had any issues. But I can't really tell what the problem your are facing is but the repo has very heavy load macros I would suspect that your computer couldn't even finish processing macros and therefore you couldn't have access to features like the ones you mentioned. I suggest you keep an eye on the left bottom corner in VSCode to follow what rust-analyzer is doing. If I am right you will see sth like "processing this and that macros".

**Reply by perfogic** [reply]
rust-analyzer finish its process but it only worked for all other files without file in tests folder and src/bin. I don't know why XD

**Reply by perfogic** [reply]
but actually this repo is really heavy, running test on it my laptop with 16gb ram sometimes hang
